pkg/apis/eksctl.io/v1alpha5/types.go

```diff
@@ -296,6 +302,14 @@ func SupportedNodeVolumeTypes() []string {
 	}
 }
 
+// SupportedSpotAllocationStrategies are the spot allocation strategies supported by ASG
+func SupportedSpotAllocationStrategies() []string {
```

Comment:
Since the only use we have for this is to check if the allocation strategy string is in the list of known values  I would instead implement a function that check's that:

```
func isSpotAllocationStrategySupported(allocationStragegy string) bool {
  switch (allocationStrategy) {
    case SpotAllocationStrategyLowestPrice, SpotAllocationStrategyCapacityOptimized:
      return true
    default:
      return false
  }
}
```

Reply:
I like having the isSupported helper, but I still like the supported strategies slice. It's currently used in the error message to print the support strategies. I used slice.Contains to implement the isSupported helper rather than the switch. When other spot allocation strategies are added to the ASG API, it will make it really easy to add the new constant to the slice and not having to touch anything else. Let me know if you agree or not.

Reply:
Sounds great :+1: